packages/core/content-manager/server/controllers/single-types.js

```diff
@@ -174,4 +174,28 @@ module.exports = {
 
     ctx.body = await permissionChecker.sanitizeOutput(unpublishedEntity);
   },
+
+  async getNumberOfDraftRelations(ctx) {
+    const { userAbility } = ctx.state;
+    const { model, id } = ctx.params;
+
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.read()) {
+      return ctx.forbidden();
+    }
+
+    const entity = await entityManager.findOneWithCreatorRolesAndCount(id, model);
+
+    if (!entity) {
+      return ctx.notFound();
+    }
+
+    if (permissionChecker.cannot.read(entity)) {
+      return ctx.forbidden();
+    }
+
+    return entityManager.getNumberOfDraftRelations(id, model);
+  },
```

Comment:
We could then change the way we get the single type entity and pass in `entity.id` to the entityManager?
```suggestion
  async getNumberOfDraftRelations(ctx) {
    const { userAbility } = ctx.state;
    const { model } = ctx.params;

    const entityManager = getService('entity-manager');
    const permissionChecker = getService('permission-checker').create({ userAbility, model });

    if (permissionChecker.cannot.read()) {
      return ctx.forbidden();
    }

    const entity = await findEntity({}, model);
    if (!entity) {
      return ctx.notFound();
    }

    if (permissionChecker.cannot.read(entity)) {
      return ctx.forbidden();
    }

    return entityManager.getNumberOfDraftRelations(entity.id, model);
  },
```